language/move-ir/types/src/location.rs

```diff
@@ -16,46 +16,53 @@ use std::{
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub struct Loc {
     file: &'static str,
-    span: Span,
+    start: u32,
```

Comment:
I kind of like how `codespan::Span` defines an index type. In this pull request there's one or two spots that replace `codespan::ByteIndex` with `u32`, and I feel like doing so makes the codebase slightly less expressive: the `u32` type being used here is specifically to represent an index into a file less than 4GB in size. If we decide later to use `u64`, we'll have to find all the spots where `u32` is being used for that purpose and change them. I think it'd be nicer to instead define `type Index = u32` (or `type ByteIndex`, I don't have strong opinions about the name) in this file, and so changing the type would be a one-line change instead of a find-and-replace.

Reply:
I can do `type ByteIndex` if you want, but in general, I really hated the whole `Span::new(ByteIndex(start), ByteIndex(end))` nonsense. It was really intrusive with no benefit. 

A type alias might be nice just for intended usage though. Thoughts on that versus just a comment?

Reply:
> I really hated the whole Span::new(ByteIndex(start), ByteIndex(end)) nonsense. It was really intrusive with no benefit.

I think the explicit constructors were necessary because `ByteIndex` was a single-element tuple made up of `RawIndex`, which was a simple type alias `type RawIndex = u32`. So if you added `type ByteIndex = u32` here, we won't have to do any of that nonsense, but still get the benefit of signaling the intended usage:

```rust
type Index = u32;
struct Span { start: Index, end: Index }
// ...
let x: u32 = 0;
let y: Index = x; // no explicit `Index(x)` necessary
Span { start: x, end: y }; // same
```

Reply:
I don't think having a span separate from the Loc is necessary. But I have updated to use a type alias